lib/buffer.js

```diff
@@ -781,34 +781,65 @@ function bidirectionalIndexOf(buffer, val, byteOffset, encoding, dir) {
 }
 
 function slowIndexOf(buffer, val, byteOffset, encoding, dir) {
-  let loweredCase = false;
-  for (;;) {
-    switch (encoding) {
-      case 'utf8':
-      case 'utf-8':
-      case 'ucs2':
-      case 'ucs-2':
-      case 'utf16le':
-      case 'utf-16le':
-      case 'latin1':
-      case 'binary':
+  switch (encoding.length) {
+    case 4:
+      if (encoding === 'utf8' || encoding === 'ucs2')
         return indexOfString(buffer, val, byteOffset, encoding, dir);
-
-      case 'base64':
-      case 'ascii':
-      case 'hex':
+      encoding = encoding.toLowerCase();
+      if (encoding === 'utf8' || encoding === 'ucs2')
```

Comment:
Instead of running `toLowerCase()` right away, it's faster to check for the all caps version first (that's more common than mixed versions).

E.g., some code from `normalizeEncoding()`.

```js
    case 5:
      if (enc === 'ascii') return 'ascii';
      if (enc === 'ucs-2') return 'utf16le';
      if (enc === 'UTF-8') return 'utf8';
      if (enc === 'ASCII') return 'ascii';
      if (enc === 'UCS-2') return 'utf16le';
      enc = `${enc}`.toLowerCase();
      if (enc === 'utf-8') return 'utf8';
      if (enc === 'ascii') return 'ascii';
      if (enc === 'ucs-2') return 'utf16le';
      break;
```

I personally believe the priority for the mixed and not string cases is so low that this would likely be even better:

```js
    case 5:
      if (enc === 'ascii') return 'ascii';
      if (enc === 'ucs-2') return 'utf16le';
      if (enc === 'UTF-8') return 'utf8';
      if (enc === 'ASCII') return 'ascii';
      if (enc === 'UCS-2') return 'utf16le';
      break;
   // ... other cases
  } // End of switch
  const lowerEnc = ('' + enc).toLowerCase();
  if (lowerEnc !== enc)
    return slowIndexOf(lowerEnc);
  throw new ERR_UNKNOWN_ENCODING(encoding);
```

Reply:
Thanks, I'll have a try.